### PR TITLE
Eslint rules

### PR DESCRIPTION
**Pull request by viktorsaws**
## Problem
We recommend to develop language servers with cross-platform compatibility between NodeJS and Web runtimes targets in mind. We don't have mechanisms to verify compatibility during development cycle.

## Solution

Setup reference eslint rules in Hello World and CodeWhisperer servers, showing how to configure linting to detect usage of platform-specific dependencies.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,10 @@
         "@types/node": "^20.12.2",
         "@typescript-eslint/eslint-plugin": "^7.7.1",
         "@typescript-eslint/parser": "^7.7.1",
+        "dependency-cruiser": "^16.3.3",
```
Comment (ege0zcan) [inline package.json:36]: is this inteded or is it leftover from testing this tool?

File: server/aws-lsp-codewhisperer/package.json [hunk 2]
```diff
@@ -21,9 +21,12 @@
         "compile": "tsc --build",
         "postcompile": "npm run copyServiceClient",
         "copyServiceClient": "copyfiles -u 1 --error ./src/client/sigv4/*.json out && copyfiles -u 1 --error ./src/client/token/*.json out",
+        "lint": "npm run lint:src",
+        "lint:bundle:webworker": "webpack --config webpack.lint.config.js && eslint bundle/aws-lsp-codewhisperer-webworker.js # Verify compatibility with web runtime target",
```
Comment (ege0zcan) [inline server/aws-lsp-codewhisperer/package.json:25]: is `lint:bundle:webworker` supposed to be run only manually?

File: server/aws-lsp-codewhisperer/package.json [hunk 3]
```diff
@@ -45,9 +48,12 @@
         "@types/uuid": "^9.0.8",
         "assert": "^2.1.0",
         "copyfiles": "^2.4.1",
+        "ts-loader": "^9.4.4",
         "ts-mocha": "^10.0.0",
         "ts-sinon": "^2.0.2",
-        "vscode-languageserver-textdocument": "^1.0.11"
+        "vscode-languageserver-textdocument": "^1.0.11",
+        "webpack-cli": "^5.1.4",
```
Comment (ege0zcan) [inline server/aws-lsp-codewhisperer/package.json:55]: shouldn't they be dev dependencies?

File: server/hello-world-lsp/webpack.lint.config.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+var path = require('path')
+
+const baseConfig = {
```
Comment (ege0zcan) [inline server/hello-world-lsp/webpack.lint.config.js:3]: it seems these configs are very similar across subpacakges, would it make sense to create a root level config and share it across the subpackages

File: server/aws-lsp-codewhisperer/package.json (hunk 2 above)
Comment (viktorsaws) [reply server/aws-lsp-codewhisperer/package.json:25]: for now yes. Adding it to any monorepo-level scripts will cause too many errors, potentially blocking other developers. For now this is a convenience tool at development time. We can extend this to release time, I may follow up on that.

File: server/aws-lsp-codewhisperer/package.json (hunk 3 above)
Comment (viktorsaws) [reply server/aws-lsp-codewhisperer/package.json:55]: yes, good catch

File: server/hello-world-lsp/webpack.lint.config.js (hunk 4 above)
Comment (viktorsaws) [reply server/hello-world-lsp/webpack.lint.config.js:3]: Choice of bundler as Webpack is very specific per package, I am not ready to introduce it as a monorepo-level feature just yet. This reference configuration can be modified per subpackage based on needs

File: server/aws-lsp-codewhisperer/package.json (hunk 3 above)
Comment (ege0zcan) [reply server/aws-lsp-codewhisperer/package.json:55]: yeah my bad, github confused me

**Review by ege0zcan — APPROVED** [review]

**Pull request merged**